### DISCUSSION TITLE
connect signal too late?

### DISCUSSION THREAD

**Discussion by octaeder** [texstudio-org/texstudio#3446] [Q&A]
I'm looking at the following code:
```c++
    QNetworkReply *reply = networkManager->get(request);
    connect(reply, SIGNAL(finished()), SLOT(onRequestCompleted()));
    connect(reply, SIGNAL(error(QNetworkReply::NetworkError)), SLOT(onRequestError()));
```
When I assume that `get(request)` immediately sends error signal before the connect of the signal is established then we would miss the error. Is the conclusion right that (for good coding) the connect should be established first before the request is send, say like this:
```c++
    QNetworkReply *reply;
    connect(reply, SIGNAL(finished()), SLOT(onRequestCompleted()));
    connect(reply, SIGNAL(error(QNetworkReply::NetworkError)), SLOT(onRequestError()));
    reply = networkManager->get(request);
```
Thx

**Comment by sunderme** [comment]
qt examples do it the same way.
